bitriseclient/certificate.go

```diff
@@ -0,0 +1,391 @@
+	IsProtected    bool   `json:"is_protected"`
+	UploadURL      string `json:"upload_url"`
+}
+
+// RegisterIdentityResponse ...
+type RegisterIdentityResponse struct {
+	Data RegisterIdentityResponseData `json:"data"`
+}
+
+// ConfirmIdentityUploadResponseData ...
+type ConfirmIdentityUploadResponseData struct {
+	UploadFileName      string `json:"upload_file_name"`
+	UploadFileSize      int    `json:"upload_file_size"`
+	Slug                string `json:"slug"`
+	Processed           bool   `json:"processed"`
+	CertificatePassword string `json:"certificate_password"`
+	IsExpose            bool   `json:"is_expose"`
+	IsProtected         bool   `json:"dais_protectedta"`
+}
+
+// ConfirmIdentityUploadResponse ...
+type ConfirmIdentityUploadResponse struct {
+	Data ConfirmIdentityUploadResponseData `json:"data"`
+}
+
+// FetchUploadedIdentityListResponseData ...
+type FetchUploadedIdentityListResponseData struct {
+	UploadFileName      string `json:"upload_file_name"`
+	UploadFileSize      int    `json:"upload_file_size"`
+	Slug                string `json:"slug"`
+	Processed           bool   `json:"processed"`
+	CertificatePassword string `json:"certificate_password"`
+	IsExpose            bool   `json:"is_expose"`
+	IsProtected         bool   `json:"dais_protectedta"`
+}
+
+// FetchUploadedIdentityListResponse ...
+type FetchUploadedIdentityListResponse struct {
+	Data []FetchUploadedIdentityListResponseData `json:"data"`
+}
+
+// FetchUploadedIdentityResponseData ...
+type FetchUploadedIdentityResponseData struct {
+	UploadFileName      string `json:"upload_file_name"`
+	UploadFileSize      int    `json:"upload_file_size"`
+	Slug                string `json:"slug"`
+	Processed           bool   `json:"processed"`
+	CertificatePassword string `json:"certificate_password"`
+	IsExpose            bool   `json:"is_expose"`
+	IsProtected         bool   `json:"dais_protectedta"`
+	DownloadURL         string `json:"download_url"`
+}
+
+// FetchUploadedIdentityResponse ...
+type FetchUploadedIdentityResponse struct {
+	Data FetchUploadedIdentityResponseData `json:"data"`
+}
+
+// FetchUploadedIdentities ...
+func (client *BitriseClient) FetchUploadedIdentities() ([]FetchUploadedIdentityListResponseData, error) {
+	log.Debugf("\nDownloading provisioning profile list from Bitrise...")
+
+	requestURL, err := urlutil.Join(baseURL, appsEndPoint, client.selectedAppSlug, certificatesEndPoint)
+	if err != nil {
+		return []FetchUploadedIdentityListResponseData{}, err
+	}
+
+	request, err := createRequest(http.MethodGet, requestURL, client.headers, nil)
+	if err != nil {
+		return []FetchUploadedIdentityListResponseData{}, err
+	}
+	log.Debugf("\nRequest URL: %s", requestURL)
+
+	// Response struct
+	requestResponse := FetchUploadedIdentityListResponse{}
+	responseStatusCode := -1
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
```

Comment:
the thing is: if the request returns non success status code, the erro will be nil.
later you do not check the status code, probably the json marshal would fail, since in case of error the response is something else then the success model, but fail fast!

please check the status code and retry if non success received

Reply:
You are right, but the performRequest do this already. If the status code != statusOK it will return an error.